### ISSUE TITLE
Can not update Binary resource

### ISSUE THREAD

**Issue by petromykhailysyn** [jamesagnew/hapi-fhir#438]
Hi @jamesagnew ,
While updating binary resource I got an error:
ca.uhn.fhir.rest.server.exceptions.InvalidRequestException: HTTP 400 Bad Request: Can not update resource, resource body must contain an ID element for update (PUT) operation
Test code:
        byte[] arr = {23,21,74,123,-44};
        Binary binary = new Binary();
        binary.setContent(arr);
        binary.setContentType("dansk");
        binary.addUndeclaredExtension(true, "bobobo", new StringDt("hey there"));

```
    IIdType resource = client.create().resource(binary).execute().getId();
    Binary fromDB = client.read().resource(Binary.class).withId(resource.toVersionless()).execute();
    IParser parser = FhirContext.forDstu2().newXmlParser();
    String xml = parser.encodeResourceToString(binary);
    System.out.println(xml);
    fromDB.setContentType("test");

    client.update().resource(fromDB).execute();
```

Resource has ID: &lt;Binary xmlns="http://hl7.org/fhir"&gt; &lt;id value=&quot;92f0f381-2d61-b53d-ba9f-6b72ee4dc31d&quot;/&gt;&lt;contentType value=&quot;dansk&quot;/&gt;&lt;content value=&quot;FxVKe9Q=&quot;/&gt;&lt;/Binary&gt;

/Petro

**Comment by RJFoster** [comment]
Which version of HAPI are you using? Some binary related stuff was fixed in 1.5
https://github.com/jamesagnew/hapi-fhir/commit/d383b402d1a46a25c3ee023246753fd1304b0c64

**Comment by petromykhailysyn** [comment]
Hello, we are using version 1.6

**Comment by jamesagnew** [comment]
Hi Petro,

Are you able to try this with the latest 2.1 SNAPSHOT? I've added a test and I'm not able to replicate this bug, so I believe it may already be fixed.

**Comment by jamesagnew** [comment]
I believe this has been corrected as a part of the fix for #521. Please feel free to reopen if are still experiencing this bug!
